### PR TITLE
Refs [TASK] [34897] [Init package,base project apply MVP]

### PR DESCRIPTION
**Pull request by manhhung23399**
## Related Tickets
- [#Task 34897: Init package,base project apply MVP](https://edu-redmine.sun-asterisk.vn/issues/34897)
## WHAT
- Create base for project
- Add library needed for project
- Create logo, icon, theme
- Create bottom navigation
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -42,4 +43,5 @@ dependencies {
     testImplementation 'junit:junit:4.+'
     androidTestImplementation 'androidx.test.ext:junit:1.1.2'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+    implementation 'com.squareup.picasso:picasso:2.71828'
```
Comment (huytq-2422) [inline app/build.gradle]: Dùng Glide thôi em

File: app/src/main/java/com/example/moviews/base/BaseFragment.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.fragment.app.Fragment
+
+abstract class BaseFragment : Fragment() {
+
+    @get: LayoutRes
+    protected abstract val layoutID: Int
+    protected abstract fun initComponent()
```
Comment (huytq-2422) [inline app/src/main/java/com/example/moviews/base/BaseFragment.kt]:
```suggestion
    protected abstract val layoutID: Int
    
    protected abstract fun initComponent()
```

File: app/src/main/res/layout/fragment_favorite.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/text_your_favorites_movie"
+            android:textColor="@android:color/white"
+            android:textSize="@dimen/sp_22"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_favorite.xml]: trải phẳng ra em ơi, hạn chế dùng constraint layout lồng nhau

File: app/src/main/res/layout/fragment_favorite.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        android:layout_height="0dp"
+        android:layout_marginTop="@dimen/dp_20"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/constraintLayoutToolbar">
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/recyclerView"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" />
+    </ScrollView>
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_favorite.xml]: tại sao lại phải lồng REcyclerview trong Scroll View

File: app/src/main/res/layout/fragment_home.xml [hunk 5]
```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/black">
+
+    <ScrollView
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_home.xml]: Sao phải gói ScrollView trong COnstraintLayout thế ?

File: app/src/main/res/layout/fragment_home.xml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                android:layout_marginTop="@dimen/dp_10"
+                android:text="@string/text_now_playing"
+                android:textColor="@android:color/white"
+                android:textSize="@dimen/sp_22"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/recyclerViewUpcoming" />
+
+            <androidx.recyclerview.widget.RecyclerView
+                android:id="@+id/recyclerViewNowPlaying"
+                android:layout_width="match_parent"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_home.xml]: Dùng match_constraint không dùng match_parent

File: app/src/main/res/layout/fragment_search.xml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/dp_16"
+        android:layout_marginEnd="@dimen/dp_16"
+        android:src="@drawable/ic_search"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerGenres"
+        android:layout_width="match_parent"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_search.xml]: match_constraint

File: app/src/main/java/com/example/moviews/base/BaseFragment.kt (hunk 2 above)
Comment (thonc-2135) [inline app/src/main/java/com/example/moviews/base/BaseFragment.kt]:
```suggestion
    protected abstract fun initComponents()
```

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1,3 +1,14 @@
 <resources>
     <string name="app_name">MoViews</string>
+    <string name="title_home">Home</string>
+    <string name="title_search">Search</string>
+    <string name="title_favorites">Favorites</string>
+    <string name="title_infomation">Information</string>
+    <string name="text_trending_now">Trending Now</string>
+    <string name="text_upcoming">Upcoming</string>
+    <string name="text_now_playing">Now Playing</string>
+    <string name="text_top_searchs">Top Searchs</string>
```
Comment (thonc-2135) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="text_top_searchs">Top Searches</string>
```

File: app/src/main/res/layout/fragment_favorite.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        android:layout_margin="@dimen/dp_16"
+        android:text="@string/text_your_favorites_movie"
+        android:textColor="@android:color/white"
+        android:textSize="@dimen/sp_22"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerView"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_favorite.xml]:
```suggestion
        android:id="@+id/recyclerXYZ"
```
ghi rõ ra được không

File: app/src/main/res/layout/fragment_home.xml [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/black">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <androidx.viewpager.widget.ViewPager
+            android:id="@+id/imgBanner"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_home.xml]: sao id lại là img thế này

File: app/src/main/java/com/example/moviews/MainActivity.kt [hunk 11]
```diff
@@ -2,10 +2,18 @@ package com.example.moviews
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.fragment.app.FragmentManager
+import com.example.moviews.screen.favorite.FavoriteFragment
+import com.example.moviews.screen.home.HomeFragment
+import com.example.moviews.screen.information.InformationFragment
+import com.example.moviews.screen.search.SearchFragment
```
Comment (KhacTung-Shjn) [inline app/src/main/java/com/example/moviews/MainActivity.kt]: không dùng mấy cái này thì import vào làm gì thế ?

File: app/src/main/res/drawable/ic_back.xml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<vector android:autoMirrored="true" android:height="24dp"
```
Comment (KhacTung-Shjn) [inline app/src/main/res/drawable/ic_back.xml]: format code đi nhé : Ctrl + Alt + L
(tất cả các file icon nhé)

File: app/src/main/res/layout/fragment_home.xml [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+            android:layout_marginStart="@dimen/dp_20"
+            android:layout_marginTop="@dimen/dp_10"
+            android:text="@string/text_trending_now"
+            android:textColor="@android:color/white"
+            android:textSize="@dimen/sp_22"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/imgBanner" />
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/recyclerViewTrendingNow"
```
Comment (KhacTung-Shjn) [inline app/src/main/res/layout/fragment_home.xml]: recyclerTrendingNow thôi nha

File: app/src/main/res/layout/fragment_home.xml [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/dp_10"
+            android:background="@color/color_drak_gray_light2"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/textNowPlaying" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+</ScrollView>
+
```
Comment (KhacTung-Shjn) [inline app/src/main/res/layout/fragment_home.xml]: remove line

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 15]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
-    <background android:drawable="@drawable/ic_launcher_background" />
-    <foreground android:drawable="@drawable/ic_launcher_foreground" />
-</adaptive-icon>
+    <background android:drawable="@color/ic_launcher_background"/>
+    <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
+</adaptive-icon>
```
Comment (KhacTung-Shjn) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml]: end line này

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -1,3 +1,15 @@
 <resources>
     <string name="app_name">MoViews</string>
+    <string name="title_home">Home</string>
+    <string name="title_search">Search</string>
+    <string name="title_favorites">Favorites</string>
+    <string name="title_infomation">Information</string>
+    <string name="text_trending_now">Trending Now</string>
```
Comment (KhacTung-Shjn) [inline app/src/main/res/values/strings.xml:9]: tách rõ title với text ra nhé nhìn cho rõ

**Pull request merged**